packages/util/src/signature.ts

```diff
@@ -39,6 +39,21 @@ function calculateSigRecovery(v: bigint, chainId?: bigint): bigint {
   return v - (chainId * BigInt(2) + BigInt(35))
 }
 
+export function calculateSigRecoveryFromV(v: bigint) {
```

Comment:
Totally a fan of having this exported in this new form, seems pretty useful to me, but can we then (also generally makes sense) some explicit test cases for this method itself, atm all tests are implicit (another method using this function is tested)?

So some edge cases like 1 (I guess we at some point decided/agreed that 0 is not a practical/used chain ID which we can ignore), a mid range even and odd value, and (related to the other comment) a very high v exceeding MAX_SAFE_INTEGER.